vignettes/philosophy.Rmd

```diff
@@ -0,0 +1,163 @@
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Everybody develops their own coding habits and style.
+Some people take a lot of effort in making their source code [readable](https://en.wikipedia.org/wiki/Computer_programming#Readability_of_source_code), while others don't bother at all.
+Working together with different people is easier when everyone uses the same standard.
+The `checklist` package defines a set of standards and provides tools to valid whether your project or R package adheres to these standards.
+You can run these tools interactive on your machine.
+You can also add this checks as [GitHub actions](https://github.com/features/actions), which runs then checks automatically after every [push](https://github.com/git-guides/git-push) to the repository on [GitHub](https://github.com).
+
+## Coding style
+
+The most visible part of a coding style is the naming convention.
+People use many different styles of naming convention within the R ecosystem [@Baath2012].
+Popular ones are `alllowercase`, `period.separated`, `underscore_separated`, `lowerCamelCase` and `UpperCamelCase`.
+We picked `underscore_separated` because that is the naming covention of the [`tidyverse`](https://tidyverse.org) packages.
+It is also the default setting in the [`lintr`](https://github.com/jimhester/lintr/blob/master/README.md) package which we use to do the [static code analysis](https://en.wikipedia.org/wiki/Static_program_analysis).
+
+At first this seems a lot to memorise.
+RStudio make things easier when you activate all diagnostic options (_Tools_ > _Global options_ > _Code_ > _Diagnostics_).
+This highlights several problems by showing a squiggly line and/or a warning icon at the line number.
+Instead of learning all the rules by heart, run `check_lintr()` regularly and fix any issues that come up.
+Do this when working on every single project.
+Doing so enforces you to consistenly use the same coding style, making it easy to learn and use it.
+
+### Rules for coding style
+
+- `underscore_separated` names for functions, parameters and variables.
+- A line of code or comments must be no longer than 80 characters.
+  Pro tip: have RStudio display this margin in the editor.
+  _Tools_ > _Global options_ > _Code_ > _Display_ > _Show margin_
+- Object names must not be longer than 30 characters.
+- Start a new line
+    - after the pipe (`%>%`)
+    - never before but always after `{`
+    - before `}`
+- Use spaces instead of tabs.
+  Pro tip: make RStudio place 2 spaces when you hit the tab key.
+  _Tools_ > _Global options_ > _Code_ > _Editing_ > _Insert spaces for tabs_
+- Use spaces consistently
+    - Use exactly one space before and after
+        - assignments `<-`, `->`, `=`
+        - operators like `+`, `-`,`*`, `/`, ...
+    - No space before and one space after `,`
+    - One space before `(` except in `function(` (don't use `function (`)
+    - One space between `)` and `{`
+    - No space after or before `(` or `[`
```

Comment:
2 lines earlier, you mention there should be one space before `(`, and now you mention there should be no space before `(`.  Or did I interpret this wrong?  I suppose it means no space after `(` and `[` and no space before `(` and `[`.  If not, I think it should be reworded and maybe split into 2 separate items.

Reply:
updated in https://github.com/inbo/checklist/pull/38/commits/ad914e2863e00df8354424d3fff611cdce8202d5

Reply:
??? No changes on vignettes/philosophy.Rmd are made in this commit.

Reply:
Ok, found your changes, looks fine now!